### ISSUE TITLE
decorateNode is called twice

### ISSUE THREAD

**Issue by alanctkc** [ianstormtaylor/slate#2895]
#### Do you want to request a _feature_ or report a _bug_?

Bug.

#### What's the current behavior?

Each plugin's `decorateNode()` method gets called twice on each render, which causes decorations to be duplicated in the decoration list.

In lieu of a JSFiddle, this one's visible in the [placeholder example](https://www.slatejs.org/#/placeholders). Each time slate-react's `Editor` `render()` is called, `decorateNode()` in each plugin is called twice. For placeholders, this means they get rendered twice (which intensifies some font weights visually):

![double-decorations](https://user-images.githubusercontent.com/273453/60035918-9a8f5180-9673-11e9-93c8-0b63c162c89e.png)

#### What's the expected behavior?

Each plugin's `decorateNode()` method should be called only once.

I assume this was a side-effect of some changes in `Node` (or something else) in the [recent annotations work](https://github.com/ianstormtaylor/slate/pull/2747), cc @ianstormtaylor.

**Comment by linonetwo** [comment]
`renderAnnotation` also got called twice, when two annotation overlay with each other.

**Comment by ianstormtaylor** [comment]
I believe that this may be fixed by https://github.com/ianstormtaylor/slate/pull/3093, which has changed a *lot* of the logic in Slate and `slate-react` especially. I'm going to close this out, but as always, feel free to open a new issue if it persists for you. Thanks for understanding.
